csharp/ql/src/semmle/code/csharp/dataflow/internal/DataFlowPrivate.qll

```diff
@@ -297,71 +297,86 @@ private module Cached {
       any(DelegateArgumentConfiguration x).hasExprPath(_, cfn, _, call)
     } or
     TMallocNode(ControlFlow::Nodes::ElementNode cfn) { cfn.getElement() instanceof ObjectCreation } or
-    TArgumentPostCallNode(ControlFlow::Nodes::ElementNode cfn) {
+    TExprPostUpdateNode(ControlFlow::Nodes::ElementNode cfn) {
       exists(Argument a, Type t |
         a = cfn.getElement() and
         t = a.stripCasts().getType()
       |
         t instanceof RefType or
         t = any(TypeParameter tp | not tp.isValueType())
       )
-    } or
-    TStoreTargetNode(ControlFlow::Nodes::ElementNode cfn) {
+      or
       instanceFieldLikeAssign(_, _, _, cfn.getElement())
+      or
+      exists(TExprPostUpdateNode upd, FieldLikeAccess fla |
+        upd = TExprPostUpdateNode(fla.getAControlFlowNode())
+      |
+        cfn.getElement() = fla.getQualifier()
+      )
     }
 
-  /**
-   * This is the local flow predicate that's used as a building block in global
-   * data flow. It may have less flow than the `localFlowStep` predicate.
-   */
+  private predicate usesInstanceField(Ssa::Definition def) {
+    exists(Ssa::SourceVariables::FieldOrPropSourceVariable fp | fp = def.getSourceVariable() |
+      not fp.getAssignable().isStatic()
+    )
+  }
+
   cached
-  predicate simpleLocalFlowStep(Node nodeFrom, Node nodeTo) {
-    any(LocalFlow::LocalExprStepConfiguration x).hasNodePath(nodeFrom, nodeTo)
+  predicate localFlowStepImpl(Node nodeFrom, Node nodeTo, boolean simple) {
+    any(LocalFlow::LocalExprStepConfiguration x).hasNodePath(nodeFrom, nodeTo) and
+    simple = true
     or
     // Flow from SSA definition to first read
     exists(Ssa::Definition def, ControlFlow::Node cfn |
-      def = nodeFrom.(SsaDefinitionNode).getDefinition()
-    |
-      nodeTo.asExprAtNode(cfn) = def.getAFirstReadAtNode(cfn)
+      def = nodeFrom.(SsaDefinitionNode).getDefinition() and
+      nodeTo.asExprAtNode(cfn) = def.getAFirstReadAtNode(cfn) and
+      if usesInstanceField(def) then simple = false else simple = true
```

Comment:
You could extract `if usesInstanceField(def) then simple = false else simple = true` into a predicate somewhere, as this repeats 4 times.